src/main/java/com/exactpro/th2/common/schema/grpc/router/impl/DefaultGrpcRouter.java

```diff
@@ -61,7 +66,32 @@
      * @throws ClassNotFoundException if matching the service class to protobuf stub has failed
      */
     public <T> T getService(@NotNull Class<T> cls) throws ClassNotFoundException {
-        return getProxyService(cls);
+        var implementations = ServiceLoader.load(Objects.requireNonNull(cls, "Services class can not be null"))
+                .stream().collect(Collectors.toList());
+
+        if (implementations.size() > 1) {
+            throw new IllegalStateException("Can not choose implementation. Fount " + implementations.size() + " implementations");
+        }
+
+        if (implementations.isEmpty()) {
+            //FIXME: Remove in future releases
+            return getProxyService(cls);
+        }
+
+        Class<?> th2ImplClass = implementations.get(0).type();
```

Comment:
If you change the generic type for Class we won't get any warnings when we create an instance. Currectly we have 'uncheckedCast' waring on line 85
```suggestion
        Class<? extends T> th2ImplClass = implementations.get(0).type();
```